sold.cc

```diff
@@ -821,6 +822,17 @@ class Sold {
         }
     }
 
+    void ConstructFilenameToSoname() {
+        for (const auto& l : link_binaries_) {
+            if (l->name() != "" && l->soname() != "") {
+                filename_to_soname_[l->name()] = l->soname();
+                LOG(INFO) << SOLD_LOG_KEY(l->name()) << SOLD_LOG_KEY(l->soname());
+            } else {
+                LOG(WARNING) << "Empty filename or soname: " << SOLD_LOG_KEY(l->name()) << SOLD_LOG_KEY(l->soname());
```

Comment:
Is the a valid code path? Shouldn't we always have both info?

Reply:
Although shared objects have non-empty soname and filename always, executable files, such as a.out, may not have DT_SONAME fields. So, I decided to allow empty sonames only for Sold::main_binary_. See https://github.com/shinh/sold/pull/37/commits/527fbd451730916b00de9b6b96ff5c17df5157ca.